kiosk/src/Components/AppModal.tsx

```diff
@@ -0,0 +1,64 @@
+import { useContext } from "react";
+import { AppStateContext } from "../State/AppStateContext";
+import ConfirmModal from "./ConfirmModal";
+import { hideModal, postNotification } from "../State/Actions";
+import { playSoundEffect } from "../Services/SoundEffectService";
+import * as Storage from "../Services/LocalStorage";
+import { removeGame } from "../Transforms/removeGame";
+import { makeNotification } from "../Utils";
+
+export default function AppModal() {
+    const { state, dispatch } = useContext(AppStateContext);
+
+    const cancelClicked = () => {
+        playSoundEffect("select");
+        dispatch(hideModal());
+    };
+
+    const deleteConfirmClicked = () => {
+        pxt.tickEvent("kiosk.deleteGame.confirmed");
+        playSoundEffect("select");
+        deleteGame();
+        dispatch(hideModal());
+    };
+
+    const selectedGame = state.allGames.find(
+        g => g.id === state.selectedGameId
+    );
+
+    const deleteGame = () => {
+        const userAddedGames = Storage.getUserAddedGames();
+        const gameId = state.selectedGameId;
+        if (gameId && gameId in userAddedGames) {
```

Comment:
typically prefer to avoid `in` as it's got confusing / mildly unexpected semantics at times (`"toString" in {} === true`) -- prefer using `userAddedGames.hasOwnProperty(gameId)` for that reason but not a big deal either way